src/log.c

```diff
@@ -56,6 +56,40 @@ static int fd_set_block(int fd) {
 }
 #endif /* PR_USE_NONBLOCKING_LOG_OPEN */
 
+void pr_log_stacktrace(module *module, int debuglevel, const char*file, int line) {
+#if defined(HAVE_EXECINFO_H) && \
+    defined(HAVE_BACKTRACE) && \
+    defined(HAVE_BACKTRACE_SYMBOLS)
+  void *trace[PR_TUNABLE_CALLER_DEPTH];
+  char **strings;
+  int tracesz;
+
+  (void) pr_log_debug(debuglevel, "%s/%s: -----BEGIN STACK TRACE from %s/%d -----",module->name,module->module_version, file, line);
+
+  tracesz = backtrace(trace, PR_TUNABLE_CALLER_DEPTH);
+  if (tracesz < 0) {
+    (void) pr_log_debug(debuglevel, "%s/%s: backtrace(3) error: %s", module->name, module->module_version, strerror(errno));
+  }
+
+  strings = backtrace_symbols(trace, tracesz);
+  if (strings != NULL) {
+    register unsigned int i;
+
+    for (i = 1; i < tracesz; i++) {
+      (void) pr_log_debug(debuglevel, "%s/%s: [%u] %s", module->name, module->module_version, i-1, strings[i]);
```

Comment:
I'd suggest doing something like this, _before_ the `for` loop:

    const char *module_name = NULL, *module_version = NULL;

    if (module != NULL) {
      module_name = module->name;
      module_version = module->module_version;
    }

Because there _will_ be cases where the callers do not have a `module *` to provide (_.e.g._ in the core API functions).

Then, in the `for` loop, you might then use:

    if (module != NULL) {
      pr_log_debug(debuglevel, "%s/%s: [%u] %s", module_name, module_version, i-1, strings[i]);

    } else {
      pr_log_debug(debuglevel, "core: [%u] %s"", i-1, strings[i]);
    }

Without these defensive checks, this code could segfault on a null pointer dereference.